gno.land/pkg/gnoland/events.go

```diff
@@ -0,0 +1,57 @@
+package gnoland
+
+import (
+	"github.com/gnolang/gno/tm2/pkg/events"
+	"github.com/rs/xid"
+)
+
+// eventType encompasses all event types
+// that can appear in the collector
+type eventType interface {
+	validatorUpdate
```

Comment:
I don't understand the meaning of this.

Reply:
You can create a collector for a specific type of event:

```go
// filterFn is the filter method for incoming events
type filterFn[T eventType] func(events.Event) []T

// collector is the generic in-memory event collector
type collector[T eventType] struct {
	events []T         // temporary event storage
	filter filterFn[T] // method used for filtering events
}
```

Instead of specifying `any` (any event allowed in the collector), I limited it to a select event types (currently only one, but there'd be more in the future, depending on what events we want to parse with the collector).

I realized when I was writing out this explanation that it was overkill, so I dropped this limitation:

60cfdbf

Reply:
I think this is better now, but I'm still considering why the collector is needed. After all, the only thing you're interested in is filtering the received events, which is done by the filterFn, right?

Reply:
I've tried assembling my thoughts on why it's needed:
https://github.com/gnolang/gno/pull/2229#discussion_r1667337494